frontend/src/metabase/admin/databases/containers/DatabaseEditApp.tsx

```diff
@@ -0,0 +1,213 @@
+/* eslint-disable react/prop-types */
+import React, { ComponentType, useState } from "react";
+import { connect } from "react-redux";
+
+import { t } from "ttag";
+import _ from "underscore";
+import { updateIn } from "icepick";
+
+import { useMount } from "react-use";
+import title from "metabase/hoc/Title";
+
+import Breadcrumbs from "metabase/components/Breadcrumbs";
+import Sidebar from "metabase/admin/databases/components/DatabaseEditApp/Sidebar/Sidebar";
+import { getUserIsAdmin } from "metabase/selectors/user";
+
+import { getSetting } from "metabase/selectors/settings";
+
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
+import DatabaseForm from "metabase/databases/containers/DatabaseForm";
+import ErrorBoundary from "metabase/ErrorBoundary";
+import { GenericError } from "metabase/containers/ErrorPages";
+import {
+  Database as DatabaseType,
+  DatabaseData,
+  DatabaseId,
+} from "metabase-types/api";
+import { State } from "metabase-types/store";
+import useBeforeUnload from "metabase/hooks/use-before-unload";
+import Database from "metabase-lib/metadata/Database";
+
+import { getEditingDatabase, getInitializeError } from "../selectors";
+
+import {
+  reset,
+  initializeDatabase,
+  saveDatabase,
+  updateDatabase,
+  syncDatabaseSchema,
+  dismissSyncSpinner,
+  rescanDatabaseFields,
+  discardSavedFieldValues,
+  deleteDatabase,
+  selectEngine,
+} from "../database";
+import {
+  DatabaseEditContent,
+  DatabaseEditForm,
+  DatabaseEditHelp,
+  DatabaseEditMain,
+  DatabaseEditRoot,
+} from "./DatabaseEditApp.styled";
+
+interface DatabaseEditAppProps {
+  database: Database & DatabaseData;
```

Comment:
Think [`Database` already extends `DatabaseData`](https://github.com/metabase/metabase/blob/7b034eb869d9aa0958e228f423c0627415dfb58c/frontend/src/metabase-types/api/database.ts#L34)?

Reply:
Yep, I saw that but I think it's an issue with the two different `Database` types that we have - this one is super weird because the `Sidebar` component accepts the old `Database` type that doesn't extend `DatabaseData`, while the `DatabaseForm` accepts the new `DatabaseData` object

Reply:
Do you mean the type from `metabase-lib/metadata/Database`? If so, I think the preferred way to fix it is to add missing property definitions to the metabase-lib type